examples/never-type/never_type.bal

```diff
@@ -0,0 +1,44 @@
+import ballerina/io;
+import ballerina/lang.'xml as xmllib;
+
+type Person record {
+    string name;
+    int age;
+};
+
+type Student record {
+    string name;
+    // You can specify an optional field with `never` type, but can not assign a value.
```

Comment:
How about something like

```suggestion
    // An optional field can be of type `never`, but a value can never be assigned to such a field.
    // This can be used to ensure that a value of the particular record type will never have a field 
    // by that name.
```

@pubudu91 wdyt?

Reply:
+1

Reply:
Ack